src/main/java/com/gladguys/alucoapi/repositories/customs/impl/CustomLessonPlanRepositoryImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.gladguys.alucoapi.repositories.customs.impl;
+
+import com.gladguys.alucoapi.entities.dto.LessonPlanDTO;
+import com.gladguys.alucoapi.entities.filters.LessonPlanFilter;
+import com.gladguys.alucoapi.repositories.customs.CustomLessonPlanRepository;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class CustomLessonPlanRepositoryImpl implements CustomLessonPlanRepository {
+
+	private JdbcTemplate jdbcTemplate;
+
+	public CustomLessonPlanRepositoryImpl(JdbcTemplate jdbcTemplate) {
+		this.jdbcTemplate = jdbcTemplate;
+	}
+
+	@Override
+	public List<LessonPlanDTO> getAllByFilters(LessonPlanFilter filter) {
+		StringBuilder sql = new StringBuilder();
+		sql.append(" SELECT l.id, l.content, l.lesson_date as lessonDate ");
+		sql.append(" FROM lesson_plan l ");
+		sql.append(" WHERE 1 = 1 ");
```

Comment:
"WHERE l.id IS NOT NULL", Não ficaria um código mais bonito de ser ler, mesmo que ambos resultem na mesma coisa?